lib/enums/query/props/flavor_properties.py

```diff
@@ -0,0 +1,68 @@
+from enum import auto
+from enums.query.props.prop_enum import PropEnum
+from exceptions.parse_query_error import ParseQueryError
+from exceptions.query_property_mapping_error import QueryPropertyMappingError
+
+
+class FlavorProperties(PropEnum):
+    """
+    An enum class for all flavor properties
+    """
+
+    FLAVOR_DESCRIPTION = auto()
+    FLAVOR_DISK = auto()
+    FLAVOR_EPHEMERAL = auto()
+    FLAVOR_ID = auto()
+    FLAVOR_IS_DISABLED = auto()
+    FLAVOR_IS_PUBLIC = auto()
+    FLAVOR_NAME = auto()
+    FLAVOR_RAM = auto()
+    FLAVOR_SWAP = auto()
+    FLAVOR_VCPU = auto()
+
+    @staticmethod
+    def from_string(val: str):
+        """
+        Converts a given string in a case-insensitive way to the enum values
+        """
+        try:
+            return FlavorProperties[val.upper()]
+        except KeyError as err:
+            raise ParseQueryError(
+                f"Could not find Flavor Property {val}. "
+                f"Available properties are {','.join([prop.name for prop in FlavorProperties])}"
+            ) from err
```

Comment:
Can we move this into the base class? As we're now up to three copies of the same method with just a slightly different error

Reply:
https://github.com/stfc/st2-cloud-pack/pull/122 - you want to basically me to do this right? I'll make a new PR